packages/wonder-stuff-i18n/src/plugins/i18n-plugin.js

```diff
@@ -0,0 +1,410 @@
+                        content: translatedSource,
+                        oldHash,
+                        hashMap: hashMap[locale],
+                        locale,
+                        getLocalePath,
+                    });
+                }
+
+                // If the contents of the file have been translated, then we
+                // need to recompute the hash based on the new contents. This
+                // also involves updating the hash in the filename.
+                if (translatedSource !== oldSource) {
+                    const newHash = hashFileContents(translatedSource);
+                    hashMap[locale][oldHash] = newHash;
+                    translatedAssetPath = translatedAssetPath.replace(
+                        oldHash,
+                        newHash,
+                    );
+                }
+
+                // Tell Webpack to write out the translated file
+                addAsset(assets, translatedAssetPath, translatedSource);
+            }
+        }
+
+        if (timing) {
+            timing.end(timingName);
+        }
+
+        return hashMap;
+    }
+
+    /**
+     * Update manifest files so that they point to the correct location of
+     * the translated files.
+     *
+     * @param {assets: Object<string, Object>, hashMap: <string, string>} options
+     */
+    localizeManifests({assets, hashMap}: {assets: Assets, hashMap: HashMaps}) {
+        const {locales, timing, shouldLocalizeManifest, getLocalePath} =
+            this.options;
+
+        const timingName = `I18nPlugin/Translation/localize-manifests`;
+
+        if (timing) {
+            timing.start(timingName);
+        }
+
+        for (const locale of locales) {
+            this.log(
+                `Updating manifests for ${locale} ${getEmojiForLocale(locale)}`,
+            );
+
+            // We then localize any manifest or HTML files, as well
+            const files = Object.keys(assets).filter((assetName) =>
+                shouldLocalizeManifest(assetName, locale),
+            );
+            for (const fileName of files) {
+                const localizedSource = localizeFile({
+                    content: assets[fileName].source(),
+                    oldHash: null,
+                    hashMap: hashMap[locale],
+                    locale,
+                    getLocalePath,
+                });
+
+                // Rename the file to the localized version
+                addAsset(assets, fileName, localizedSource);
+            }
+        }
+
+        if (timing) {
+            timing.end(timingName);
+        }
+    }
+
+    /**
+     * Logic to run when emitting the files. This will get the translated
+     * strings, translate the assets, and update the manifest files.
+     *
+     * @param {Object} options compilation options from Webpack
+     * @param {Function} callback callback function to call when done
+     * @returns void
+     */
+    async onEmit(
+        {assets}: {assets: Assets},
```

Comment:
I wish there was a webpack class that plugins could extend so that we could get some of these types for free.  No action req'd.

Reply:
Yeah, that would be nice!